tests/functional/u/used/used_before_assignment_postponed_evaluation.py

```diff
@@ -0,0 +1,13 @@
+"""Tests for used-before-assignment when postponed evaluation of annotations is enabled"""
+# pylint: disable=missing-function-docstring, invalid-name
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    var = 1
+    import math
+
+print(var)  # [used-before-assignment]
+
+def function_one(m: math):  # no error for annotations
+    return m
```

Comment:
```suggestion
    return m


class StreamExample(MusicStream):   # [used-before-assignment]
    def my_func(self):
        if TYPE_CHECKING:
           assert isinstance(self, Stream)
           assert isinstance(self, MusicStream)  # [used-before-assignment]
```

Reply:
I thought we were deferring this to #8167 per https://github.com/pylint-dev/pylint/pull/8441#issuecomment-1489614574?

Reply:
Ho my bad did not read in detail :)